src/components/hooks/useKeyPress.js

```diff
@@ -0,0 +1,42 @@
+import {useState, useEffect} from 'react';
+
+/**
+ * Custom hook that returns wether a key is pressed on an element.
+ *
+ * @param {string} targetKey  Target key to listen for.
+ * @param {RefObject} elementRef  Element on which to listen for event.
+ * @returns {boolean} Wehter or not a key has been pressed.
```

Comment:
maybe just return the pressed key to simplify the usage, only one listener for all keys.